lib-image/src/main/java/ai/kognition/pilecv4j/image/Utils.java

```diff
@@ -1128,13 +1128,29 @@ public static Size preserveAspectRatio(final Mat mat, final Size newSize) {
         return preserveAspectRatio(mat.size(), newSize);
     }
 
+    /**
+     *
+     * @param originalMatSize The Size of the original image
+     * @param newSize The desired Size of the new image
+     * @return The size of the new image, scaled to match the height or width of {@param newSize}
```

Comment:
`scaled to match the height or width of {@param newSize}` - whichever is largest of the original image, right?

Reply:
Updated in a new commit. Unfortunately its kind of convoluted. The block of text below is supposed to break down exactly how to understand the return value. 

Alternatively, I could refactor the code to make the return value more intuitive, such as always matching the greater of the two specifications in `newsize`.